bpf/kmesh/workload/xdp.c

```diff
@@ -109,6 +96,81 @@ static inline int parser_xdp_info(struct xdp_md *ctx, struct xdp_info *info)
     return PARSER_SUCC;
 }
 
+static inline int xdp_deny_packet(struct xdp_info *info, struct bpf_sock_tuple *tuple_info)
+{
+    if (info->iph != NULL && info->iph->version == 4) {
+        BPF_LOG(
+            INFO,
+            XDP,
+            "auth denied, src ip: %s, port: %u\n",
+            ip2str(&tuple_info->ipv4.saddr, true),
+            bpf_ntohs(tuple_info->ipv4.sport));
+    } else {
+        BPF_LOG(
+            INFO,
+            XDP,
+            "auth denied, src ip: %s, port: %u\n",
+            ip2str(&tuple_info->ipv6.saddr[0], false),
+            bpf_ntohs(tuple_info->ipv6.sport));
+    }
+    return XDP_DROP;
+}
+
+static inline wl_policies_v *get_workload_policies(struct xdp_info *info, struct bpf_sock_tuple *tuple_info)
+{
+    frontend_key frontend_k = {};
+    frontend_value *frontend_v = NULL;
+    __u32 workload_uid = 0;
+
+    if (info->iph->version == 4) {
+        frontend_k.addr.ip4 = tuple_info->ipv4.daddr;
+    } else {
+        if (is_ipv4_mapped_addr(tuple_info->ipv6.daddr))
+            V4_MAPPED_REVERSE(tuple_info->ipv6.daddr);
```

Comment:
This is changing the tuple_info, if we use it later, then it will be error prone

Reply:
fix